projects/js-packages/publicize-components/src/components/social-image-generator/panel/index.js

```diff
@@ -31,6 +31,10 @@ const SocialImageGeneratorPanel = ( { prePublish = false } ) => {
 		setTemplate,
 	} = useImageGeneratorConfig();
 
+	useEffect( () => {
+		setIsEnabled( isEnabled );
+	}, [] ); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
When creating a new post or editing an existing one, the [default value for enabling SIG is already set here](https://github.com/Automattic/jetpack/blob/trunk/projects/js-packages/publicize-components/src/hooks/use-image-generator-config/index.js#L8). However, this value was not being updated in the post meta. This `useEffect` hook simply updates the meta with the current value when the SIG panel is opened, so that it gets saved properly.

ESLint is being ignored because it wants `setIsEnabled` and `isEnabled` in the dependency array, which causes an infinite loop.

The tradeoff here is that SIG won't be enabled if a post is created programatically, since the post meta won't be present. This however can be fixed fairly easily by adding the post meta when the post is being created.